manager/volume.go

```diff
@@ -232,11 +232,13 @@ func (m *VolumeManager) Attach(name, nodeID string, disableFrontend bool, attach
 		return nil, err
 	}
 
-	if isReady, err := m.ds.CheckEngineImageReadyOnAtLeastOneVolumeReplica(v.Spec.EngineImage, v.Name, nodeID); !isReady {
-		if err != nil {
-			return nil, fmt.Errorf("cannot attach volume %v with image %v: %v", v.Name, v.Spec.EngineImage, err)
+	if v.Spec.DataLocality != longhorn.DataLocalityStrictLocal {
```

Comment:
Why we need to add the condition here? Checking the engine binary on the node is quite general, so it should nto be filtered out?

Reply:
Because the replica of the local volume is not really scheduled to a node after creation, the scheduling to the engine's node is done after executing attaching. So, the `replica.spec.nodeID` is empty, `CheckEngineImageReadyOnAtLeastOneVolumeReplica` will always fail.

Reply:
I see, but you can't just bypass checking image on `nodeID`, otherwise the engine and replica on `nodeID` would not be able to start. 

So it's better to skip checking images on replica nodes in `CheckEngineImageReadyOnAtLeastOneVolumeReplica` if the volume is data-locality-strict but keep checking `nodeID`?

Reply:
Addressed. Thank you.